src/index/hnsw.cpp

```diff
@@ -800,4 +810,17 @@ HNSW::init_memory_space() {
     return true;
 }
 
+void
+HNSW::get_vectors(const vsag::DatasetPtr& base, void*& vectors, size_t& data_size) const {
+    if (type_ == DataTypes::DATA_TYPE_FLOAT) {
+        vectors = (void*)base->GetFloat32Vectors();
+        data_size = dim_ * sizeof(float);
+    } else if (type_ == DataTypes::DATA_TYPE_INT8) {
+        vectors = (void*)base->GetInt8Vectors();
+        data_size = dim_ * sizeof(int8_t);
+    } else {
+        throw std::invalid_argument("fail to support this data type");
```

Comment:
the exception message  need include unknown type id

Reply:
i removed this choice because the part will be completed during the parameter checking phase

Reply:
For this function, it is still recommended to add a check for the parameter `type_`.

Reply:
done